### ISSUE TITLE
unsupported RDATE parm: TZID=EUROPE/PRAGUE

### ISSUE THREAD

**Issue by tchap** [jakubroztocil/rrule#129]
Rule:

``` javascript
'EXDATE;TZID=Europe/Prague:20120706T123000,20130405T123000,20130510T123000,20131004T123000,20131101T123000,20131206T123000,20131213T123000,20131227T123000,20140103T123000,20150501T123000,20150508T123000'
```

Got `unsupported RDATE parm: TZID=EUROPE/PRAGUE`

**Comment by gioele-antoci** [comment]
Same issue here.
Shouldn't this be a valid string according to this (from here: [RFC 5545](https://tools.ietf.org/html/rfc5545#section-3.8.5))?
``` 
exdate     = "EXDATE" exdtparam ":" exdtval *("," exdtval) CRLF

       exdtparam  = *(
                  ;
                  ; The following are OPTIONAL,
                  ; but MUST NOT occur more than once.
                  ;
                  (";" "VALUE" "=" ("DATE-TIME" / "DATE")) /
                  ;
                  (";" tzidparam) /
                  ;
                  ; The following is OPTIONAL,
                  ; and MAY occur more than once.
                  ;
                  (";" other-param)
                  ;
                  )

       exdtval    = date-time / date
       ;Value MUST match value type
```

**Comment by marcoancona** [comment]
+1

**Comment by marcrd** [comment]
@tchap @gioele-antoci I ran into this in our own app, and here's the fix I came up with to at least filter out `EXDATE` and `RDATE` properties and make the library not destructively error out:


```diff
+    /**
+     * Filters out floating timezone properties.
+     * See https://tools.ietf.org/html/rfc5545#section-3.2.19 for more on the time zone property part of the iCal spec.
+     * See https://tools.ietf.org/html/rfc5545#section-3.3.5 for more on floating times in the iCal spec.
+     * @param {String} stringRule A string containing the set of recurrence properties.
+     * @return {String}           A filtered string containing the set of recurrence properties without floating
+     *                            timezones.
+     */
+    filterFloatingTimezones: function (stringRule) {
+      // Uses a regex to split the recurrence string by carriage return.
+      var parsedRecurrenceString = stringRule.split(/\s+/)
+
+      // Remove any of the recurrence date properties that contain floating timezone properties.
+      parsedRecurrenceString.forEach(function (ruleString) {
+        var rruleIndex = parsedRecurrenceString.indexOf(ruleString);
+
+        if (contains(ruleString, 'TZID') && rruleIndex !== -1) {
+          parsedRecurrenceString.splice(rruleIndex, 1)
+        }
+      })
+
+      return parsedRecurrenceString.join('\n')
+    },
+
      parse: function (s, options) {
        options = options || {}
  
        var invalid = []
        var keys = Object.keys(options)
        var defaultKeys = Object.keys(RRuleStr.DEFAULT_OPTIONS)
  
+      // Sets the recurrence string to the original recurrence model without any floating timezones.
+      filteredRecurrenceString = this.filterFloatingTimezones(s)
+
        keys.forEach(function (key) {
          if (!contains(defaultKeys, key)) invalid.push(key)
        }, this)
 
       if (invalid.length) throw new Error('Invalid options: ' + invalid.join(', '))
 
       // Merge in default options
       defaultKeys.forEach(function (key) {
          if (!contains(keys, key)) options[key] = RRuleStr.DEFAULT_OPTIONS[key]
        })
  
-      return this._parseRfc(s, options)
+      return this._parseRfc(filteredRecurrenceString, options)
      }
    }

```

**Comment by tchap** [comment]
@marcrd Haven't used this lib in ages, but cheers anyway 👍

**Comment by davidgoli** [comment]
This is a duplicate of #60 (and others), closing

**Comment by davidgoli** [comment]
Fixed as of 2.4.0
